cmd/tempo/app/modules.go

```diff
@@ -133,12 +135,34 @@ func (t *App) initQuerier() (services.Service, error) {
 	tracesHandler := middleware.Merge(
 		t.httpAuthMiddleware,
 	).Wrap(http.HandlerFunc(t.querier.TraceByIDHandler))
-
 	t.server.HTTP.Handle("/api/traces/{traceID}", tracesHandler)
 
 	return t.querier, nil
 }
 
+func (t *App) initQueryFrontend() (services.Service, error) {
+	var err error
+	// frontend has its own roundTripper. f.roundTripper is f
+	t.frontend, err = frontend.New(t.cfg.Frontend.Config, util.Logger, prometheus.DefaultRegisterer)
+	if err != nil {
+		return nil, err
+	}
+
+	// custom tripperware that splits requests
+	tripperware, err := querier.NewTripperware(t.cfg.Frontend, util.Logger, prometheus.DefaultRegisterer)
+	if err != nil {
+		return nil, err
+	}
+	// tripperware will be called before f.roundTripper (which calls roundtripgrpc)
+	t.frontend.Wrap(tripperware)
+
+	frontend.RegisterFrontendServer(t.server.GRPC, t.frontend)
+	// register at a different endpoint for now
+	t.server.HTTP.Handle("/api/traces/frontend/{traceID}", t.frontend.Handler())
```

Comment:
once this is merged i think i'd prefer the reverse.  the query frontend should own the primary endpoint, but we can keep a separate one to make a request against the querier directly.  something like:

`/querier/api/traces/{traceID}`

Reply:
I agree with this 100%, frontend should eventually own the primary endpoint.